dFusion/dFusion.rst

```diff
@@ -184,170 +182,230 @@ TODO
+During the auction settlement, only a subset of the orderstream is actually considered and can be settled.
+
+Any order once touched in a solution will also never be considered as a valid order, as we can not differeniate between intentially replayed orders and maliciously replayed ordres.
 
 
-Finding the batch price: optimization of batch trading surplus (off-chain)
+Finding the batch price: optimization of batch trading utility (off-chain)
 --------------------------------------------------------------------------
 
-After a certain time-frame or once the maximum number of orders per batch are collected, a batch is "frozen" and the orders participating in it are final.
+After a certain time-frame, anyone can trigger a "batch-freeze" and the a snap-shot of the latest orderstream is made.
 A new batch could immediately start collecting new orders while the previous one is being processed.
-To process a batch, participants compute the uniform clearing price maximizing the trading surplus between all trading pairs can. 
-The traders surplus of an order is defined as the difference between the uniform clearning price and the limit price, multipied by the volume of the order with respect to some reference token. 
+To process a batch, participants compute the uniform clearing price maximizing the trading utility between all trading pairs. 
+The traders utility of an order is defined as the difference between the uniform clearning price and the limit price, multipied by the volume of the order with respect to some reference token. 
 The exact procedure is described `here <https://github.com/gnosis/dex-research/blob/master/BatchAuctionOptimization/batchauctions.pdf>`_. 
-Calculating the uniform clearing prices is an np-hard optimization problem and most likely the global optimum will not be found in the pre-defined short time frame: **SolvingTime** - estimated between 3-10 minutes. 
+Calculating the uniform clearing prices is an np-hard optimization problem and most likely the global optimum will not be found in the pre-defined short time frame of 3-10 minutes.
 While we are unlikely to find a global optimum, the procedure is still fair, as everyone can submit their best solution.
-Since posting the complete solution (all prices and traded volumes) would be too gas expensive to put on-chain for each candidate solution, participants only submit the 'traders surplus' they claim there solution is able to achieve.
-The anchor contract will store all submissions and will select the solution with the maximal 'traders surplus' as the final solution.
+However, many heuristic approaches might exist to find reasonable solutions in a short timeframe.
+
+Since posting the complete solution (all prices and traded volumes) would be too gas expensive to put on-chain for each candidate solution, participants only submit the 'traders utility' they claim there solution is able to achieve, the new balance-state-hash after the auction settlement and a bond.
```

Comment:
```suggestion
Since posting the complete solution (all prices and traded volumes) would be too gas expensive to put on-chain for each candidate solution, participants only submit the 'traders utility' they claim there solution is able to achieve, the new balance-state-hash after the auction settlement along with a bond.
```